lib/global_id/identification.rb

```diff
@@ -21,5 +21,31 @@ def to_signed_global_id(options = {})
     def to_sgid_param(options = {})
       to_signed_global_id(options).to_param
     end
+
+    ActiveRecord::Associations::Builder::BelongsTo.instance_eval do
+      def define_accessors(model, reflection)
+        class_name = reflection.options.fetch(:class_name, reflection.name.capitalize.to_s)
+        foreign_key = reflection.options.fetch(:foreign_key, "#{reflection.name}_id")
+
+        if reflection.options[:polymorphic]
+          class_name = "#{reflection.name}_type"
+        end
+
+        ["#{reflection.name}_gid", "#{reflection.name}_global_id"].each do |method|
+          model.send(:define_method, method) do
+            GlobalID.create_from(reflection.options[:polymorphic] ? send(class_name) : class_name, send(foreign_key))
```

Comment:
I think this could be:

```ruby
model_name = reflection.polymorphic? ? reflection.foreign_type : reflection.class_name

# later...
GlobalID.create_from(send(model_name), send(reflection.foreign_key))
```